### PR TITLE
Checkpoint PR

### PR DESCRIPTION
**Pull request by jshilmover**
Specifications

    Create a model (Product) and migration to create a table (products)
        Fields:
            id, timestamps are autogenerated
            name
            price (use decimal datatype and assure it can hold dollars.cents, you may need to research this a bit)
            description
            item_number
            image (use imageUrl method as part of the faker library. research it a bit to understand it better)
    Use Laravel's seeding capability. Seed at least 30 products
    Verify records were created in Tinker
    Generate a resourceful controller that enables you to do CRUD actions
    Create a page that shows all products (found at: /products)
        Show a small/thumbnail image for each product- just not a full image view
        Show name, price (formatted as dollar amount), and item_number only
        Assure the index page paginates the products - 10 per page
        From the index page, each product's name should be linkable to the show page
    Create a page that shows a product (found at: /product/<id>, ie. /products/35)
        Show full/mid size image for product
        Show all data for a product
    Use a UI framework, ie. Bootstrap, to make your pages look nice and responsive

--------------------------------------------------------- Checkpoint ---------------------------------------------------------

    Push assignment/crud-app branch to Github
    Create a pull request (PR) - no need for description or images.
    Tag me as a reviewer
    I'll give you feedback. This is not being graded yet. Feedback only.
    Once I give the go-ahead then proceed with the rest of the assigned work

--------------------------------------------------------- Checkpoint ---------------------------------------------------------

 

 

    On index page, show a button to create a product
    On index page,
        For each record, you can click button/link to edit and another one to also delete
            You cannot delete a product without a JavaScript confirmation warning.
    Create and Edit Form pages
        Share common fields in a shared file
        Validate and repopulate data
            Use Laravel validation only. Do not use HTML or JavaScript validation. In the real-world you'll use both frontend and backend validation techniques.
        Do not include id or timestamps in forms
        You can leave the image field untouched. You may need to take off any requirement validation for the image field. It wouldn't be too much to use faker when creating and editing images, if you want. But that is optional.
    Validation code in the controller is defined in one place - not copy pasted for create and edit actions. Validation code should be shared for both.
    Assure common code is put into a layout file to be used for both list and show

Extra Credit (not required and only if you have time)

    When seeding, use a factory to create records. You will need to research it.
    Screen-record a video showcasing functionality of your CRUD application
        If you do this then you do NOT have to add screenshots to PR description
        Embed screen-recording onto the PR description

### CODE REVIEW

File: crud-app/resources/views/products/view_product.blade.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
+<!DOCTYPE html>
```
Comment (kblake) [inline crud-app/resources/views/products/view_product.blade.php:1]: Change the name of this file to the canonical name of `show.blade.php`

File: crud-app/app/Http/Controllers/ProductController.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        return view('products/view_product', [
```
Comment (kblake) [inline crud-app/app/Http/Controllers/ProductController.php:51]: navigate to `show` instead of `view_product`

**Review by kblake — COMMENTED** [review]
A file change tweak but other than that looks good! Please continue with rest of assignment specifications.

**Comment by jshilmover** [comment]
Index page: 
image
image

Show page:
image

Create page:
image

Edit page:
image

Delete confirmation:
image

File: crud-app/app/Http/Controllers/ProductController.php [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+     */
+    public function store(Request $request)
+    {
+        $validated = Product::validateEntry($request);
+
+        $product = new Product;
+        $product->name = $request->productName;
+        $product->price = $request->productPrice;
+        $product->item_number = $request->itemNumber;
+        $product->description = $request->description;
```
Comment (kblake) [inline crud-app/app/Http/Controllers/ProductController.php:46]: TODO: Update your form fields to match property and table field names and then you don' have to assign each one separately.

File: crud-app/app/Http/Controllers/ProductController.php [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+     */
+    public function update(Request $request, $id)
+    {
+        $validated = Product::validateEntry($request);
+
+        $product = Product::find($id);
+        $product->name = $request->productName;
+        $product->price = $request->productPrice;
+        $product->item_number = $request->itemNumber;
+        $product->description = $request->description;
```
Comment (kblake) [inline crud-app/app/Http/Controllers/ProductController.php:96]: Same here as in the store action

File: crud-app/app/Models/Product.php [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    use HasFactory;
+
+    public static
+    function validateEntry(Request $request)
+    {
+        return
+            $request->validate([
+                'productName' => 'required',
+                'productPrice' => 'required|decimal:2',
+                'itemNumber' => 'required|numeric',
```
Comment (kblake) [inline crud-app/app/Models/Product.php:21]: rename fields to match table fields

**Review by kblake — CHANGES_REQUESTED** [review]
Feedback:

- Create and leverage a layout file for all your blade templates to use. 
- You are missing an edit page.
- Rename your form element names to match table field names

**Review by kblake — APPROVED** [review]
+5 ec for factory usage
-11 for unresolved feedback

**Pull request merged**